gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -378,42 +394,53 @@ def run(self) -> GtsfmData:
         )
         all_metrics_groups = [retriever_metrics, two_view_agg_metrics]
 
-        delayed_sfm_result, delayed_io, delayed_mvo_metrics_groups = self.scene_optimizer.create_computation_graph(
-            keypoints_list=keypoints_list,
-            i2Ri1_dict=i2Ri1_dict,
-            i2Ui1_dict=i2Ui1_dict,
-            v_corr_idxs_dict=v_corr_idxs_dict,
-            two_view_reports=two_view_reports_dict,
-            num_images=len(self.loader),
-            images=self.loader.create_computation_graph_for_images(),
-            camera_intrinsics=intrinsics,
-            relative_pose_priors=self.loader.get_relative_pose_priors(image_pair_indices),
-            absolute_pose_priors=self.loader.get_absolute_pose_priors(),
-            cameras_gt=self.loader.get_gt_cameras(),
-            gt_wTi_list=self.loader.get_gt_poses(),
-            gt_scene_mesh=self.loader.get_gt_scene_trimesh(),
-        )
-
-        with performance_report(filename="scene-optimizer-dask-report.html"):
-            sfm_result, *other_results = dask.compute(delayed_sfm_result, *delayed_io, *delayed_mvo_metrics_groups)
-        mvo_metrics_groups = [x for x in other_results if isinstance(x, GtsfmMetricsGroup)]
-
-        assert isinstance(sfm_result, GtsfmData)
-        all_metrics_groups.extend(mvo_metrics_groups)
-
-        end_time = time.time()
-        duration_sec = end_time - start_time
-        logger.info("GTSFM took %.2f minutes to compute sparse multi-view result.", duration_sec / 60)
+        # Partition image pairs if a partitioner is provided
+        if graph_partitioner is not None:
+            subgraphs = graph_partitioner.partition_image_pairs(image_pairs)
+        else:
+            # If no partitioner is provided, use a single partition
+            subgraphs = [image_pairs]
+
+        # Group results by subgraph
+        subgraph_two_view_results = group_results_by_subgraph(two_view_results_dict, subgraphs)
+
+        # Process each subgraph
+        all_subgraph_results = []
+        for idx, subgraph_result_dict in enumerate(subgraph_two_view_results):
+            logger.info(f"Processing subgraph {idx+1}/{len(subgraph_two_view_results)} with {len(subgraph_result_dict)} image pairs")
+            
+            # Unzip the two-view results for this subgraph
+            subgraph_i2Ri1_dict, subgraph_i2Ui1_dict, subgraph_v_corr_idxs_dict, subgraph_pre_ba_reports, subgraph_post_isp_reports = unzip_two_view_results(
+                subgraph_result_dict
+            )
+            
+            # Run scene optimization for this subgraph
+            if len(subgraph_i2Ri1_dict) > 0:  # Only process non-empty subgraphs
+                subgraph_result, _, _ = self.scene_optimizer.create_computation_graph(
```

Comment:
we need the metrics and the IO results, please do not discard those. None of the the functionality from existing code should be lost.